### PR TITLE
#90 doc

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- tree fix selection and clean code

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
💯

File: gui/doc/toggle.md [hunk 1]
```diff
@@ -1,21 +1,106 @@
-A series of toggle buttons that the user can select.
+A serie of toggle buttons that the user can select.
```
Comment (FabienLelaquais) [inline gui/doc/toggle.md]: The proper term is 'series'... even if there's only one... serie.

File: gui/doc/toggle.md [hunk 2]
```diff
@@ -1,21 +1,106 @@
...
+Each button is represented by a string, an image or both.
+
+You can use an arbitrary type for all the items (see the [example](#binding-to-a-list-of-objects)).
 
 ## Usage
 
-### Simple
+### Display a list of string
+
+You can create a list of toggle buttons from a series of strings:
```
Comment (FabienLelaquais) [inline gui/doc/toggle.md:13]: Just like here

File: gui/doc/toggle.md [hunk 3]
```diff
@@ -1,21 +1,106 @@
...
+  
+    === "HTML"
+
+        ```html
+        <taipy:toggle lov="Item 1;Item 2;Item 3">{value}</taipy:toggle>
+        ```
+
+### Unselect value
+
+In a serie of toggle buttons, the user can unselect a button (no value is selected.
```
Comment (FabienLelaquais) [inline gui/doc/toggle.md]: serieS

The sentence is not really clear, to me.
=>
"In a `toggle` control, all buttons might be unselected. Therefore there is no value selected."

File: gui/doc/toggle.md [hunk 4]
```diff
@@ -1,21 +1,106 @@
...
+  
+    === "HTML"
+
+        ```html
+        <taipy:toggle lov="Item 1;Item 2;Item 3" unselected_value="No Value">{value}</taipy:toggle>
+        ```
+
+### Display a list of tuples
+
+A toggle control that returns an id while selecting a label or Icon^.
```
Comment (FabienLelaquais) [inline gui/doc/toggle.md]: Use backticks around Icon^ for a link to be created.

File: gui/doc/toggle.md [hunk 5]
```diff
@@ -1,21 +1,106 @@
...
-<|{value}|toggle|lov=Item 1;Item 2;Item 3|></code>
-<code><|toggle|theme|></code>
+!!! example "Page content"
 
-### Advanced
+    === "Markdown"
 
-<code><|{value}|toggle|lov={lov}|type=myType|adapter=lambda x: (x.id, x.name)|></code>
+        ```
+        <|{user_sel}|toggle|lov={users}|type=User|adapter=lambda u: (u.id, u.name)|>
```
Comment (FabienLelaquais) [inline gui/doc/toggle.md]: No curly braces for the lambda here? that would be shocking...

File: gui/doc/tree.md [hunk 6]
```diff
@@ -1,15 +1,225 @@
...
+    === "HTML"
+
+        ```html
+        <taipy:tree lov="Item 1;Item 2;Item 3" multiple="True" filter="True">{value}</taipy:tree>
+        ```
+
+
+### Display a list of tuples
+
+A tree control that returns an id while selecting a label or Icon^.
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: Use backticks for the link.

File: gui/doc/tree.md [hunk 7]
```diff
@@ -1,15 +1,225 @@
...
+    === "HTML"
+
+        ```html
+        <taipy:tree lov="Item 1;Item 2;Item 3" multiple="True" filter="True">{value}</taipy:tree>
+        ```
+
+
+### Manage expanded nodes
+
+The property _expanded_ helps to manage the state of the nodes.
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: "helps to manage the state of the nodes"
->
"must be used to control the expanded/collapse state of the nodes"

File: gui/doc/tree.md [hunk 8]
```diff
@@ -1,15 +1,225 @@
...
+By default, the user can expand or collapse nodes.
+If _expanded_ is set to False, there can be only one expanded node at any given level of the tree.
+ie if a node is expanded at a certain level and the user click on another node at the same level, the first node will be automatically collapsed.
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: I'd rather see a colon here:
"tree. ie if a node"
->
"tree: if a node"

File: gui/doc/tree.md [hunk 9]
```diff
@@ -1,15 +1,225 @@
...
+
+The _expanded_ property can also hold a list of ids that will be automatically expanded.
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: "that will be automatically expanded"
=>
"that are expanded"

File: gui/doc/tree.md [hunk 10]
```diff
@@ -1,15 +1,225 @@
...
+        <|{user_sel}|tree|lov={users}|>
+        ```
+  
+    === "HTML"
+
+        ```html
+        <taipy:tree lov="{users}" value="{user_sel}" />
+        ```
+
+### Display a list of dict with buil-in adapter
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: dict -> dictionary

built-in

File: gui/doc/tree.md [hunk 11]
```diff
@@ -1,15 +1,225 @@
...
+
+Dictionnaries with keys _id_, _label_ and _children_ (as a list) can de dealt directly by the built-in _lov_ adapter.
```
Comment (FabienLelaquais) [inline gui/doc/tree.md]: Dictionnaries -> Dictionaries

**Review by FabienLelaquais — APPROVED** [review]
Excellent.
The problem with links without backticks also appear in part.md... you will want to fix that as well.
Thanks!

File: gui/doc/toggle.md (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/doc/toggle.md]: ok

**Pull request merged**